src/Orc.CommandLine/Context/Interfaces/IResult.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace Orc.CommandLine
+{
+    using Catel.Data;
+
+    public interface IResult : IContext
```

Comment:
one of the points of the issue is to split ```IContext``` into ```IResult``` and ```ICommandLineParsingContext```
it means ```IResult``` should not be inherited from ```IContext```

I would extract two different interfaces from ```IContext```: ```ICommandLineParsingContext``` and something like ```ICommandLineParsingResult``` and do something like that:
```
[ObsoleteEx]
public interface IContext : ICommandLineParsingContext, ICommandLineParsingResult
{
    // put obsolete props and methods here
}

public interface IValidatedResult : ICommandLineParsingResult
{
   IValidationContext ValidationContext { get; }
}
```